Python/syntax_test_python.py

```diff
@@ -266,7 +266,7 @@ def _():
 #                       ^ invalid.illegal.expected-parameter.python
 #                            ^ invalid.illegal.expected-parameter.python
 
-    lambda
+    lambda x
```

Comment:
I suppose you were talking about this invalid line breaking the tests? It's still invalid, but the bare keyword is used specifically for bail-out during typing. The lambda context should be exited if a newline without a backslash follows.

Reply:
I agree that the current failure mode is pretty poor. We'd run into a similar problem here:

```python
lambda foo,
42
```

But we can't simply bail on a newline because the lambda could be inside parens:

```python
(lambda
foo
:0)
```

In order to handle errors gracefully, we might need duplicate implementations of lambdas inside and outside parens. I hope there's a better solution than that.

Reply:
Huh, I didn't even know this was possible.

It's pretty much an anti-pattern though. If you end up in a position where you'd look into multi-lining the arguments to a `lambda` function, you probably should be using `def` already. I've never seen this before and I don't think it's likely to pop up, so … I guess you're free to implement two lambda versions, but I doubt I'll bother.

Reply:
I had to try it out to be sure. Python grammar is hard (and, in principle, impossible to parse in a Sublime syntax).

Anyway, now that I think about it, when would you ever write a lambda outside parens in the first place? As long as the behavior in such a case is technically correct, it probably isn't a big deal if it's a bit ungraceful (certainly next to the hassle of duplicate code).

Reply:
Good point. Now that PEP8 (and tooling) actively discourages you from assigning lambdas (iirc it didn't do that a few years ago), you hardly ever write lambdas outside of parens and when you do, you probably don't care about 100% accutate highlighting either.

I guess we can safely ignore the line-splitting case for incorrect grammar and just bail out as soon as we find an unrecognized token. Should put a comment with this decision somewhere around the lambda match.